### DISCUSSION TITLE
Dynamic path

### DISCUSSION THREAD

**Discussion by pikachu937** [bluenviron/mediamtx#1101] [Questions]
Hello.
I have some problem.
Your rtsp-simple-server read streams from my cameras, but only if I explicitly set the path to each stream, for example:
paths:
  path1:
    source: rtsp://login:pass@ip:rtsp_port/rtsp_stream
    sourceProtocol: tcp
  path2:
    source: source: rtsp://login:pass@ip:rtsp_port/rtsp_stream
    sourceProtocol: tcp
I have a database that stores all thread addresses.

Question: How can I make rtsp-simple-server access db(mysql/pgsql/sqlite) or text file, get information and start reading them?
As an option, a sheet with paths is dynamically created (based on data received from another source, such as db or a text file) and reading is started.

Thanks

**Comment by aler9** [comment]
Hello, you can create an app written in any language you want that

1. starts the server
2. loads path data from a DB or from a text file
3. set the configuration of the server by using the [HTTP API](https://github.com/aler9/rtsp-simple-server#http-api)

**Comment by pikachu937** [comment]
Hello.  I got you right, do you suggest using an external application with a downloadable list of paths and connecting them in real time via the http api?

**Reply by aler9** [reply]
yep, that's what i do to configure instances in order to interact with ~1000 cameras stored in a DB. It's senseless to add support for a single DB or text format, since everyone wants to use a different format and a different logic. Just write things with your logic and your way of doing things, then push them to the server by using the API.

[a comment by an automated account is not shown]
